File: 1957-delete-characters-to-make-fancy-string/1957-delete-characters-to-make-fancy-string.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        res = []
        for c in s:
            if len(res) >= 2 and res[-1] == c and res[-2] == c: # 每次加入之前，檢查 res 的最後兩個字元是否都等於目前的字元
                continue
            res.append(c)
        return ''.join(res)

File: 1957-delete-characters-to-make-fancy-string/1957-delete-characters-to-make-fancy-string.cpp
class Solution {
public:
    string makeFancyString(string s) {
        string res;
        res.reserve(s.size()); // 使用 reserve 提前配置記憶體（優化效能)

        for(char c: s){
            int n = res.size();
            // 如果 res 已有至少兩個字元，且最後兩個字元都等於 c，
            // 就跳過這個字元（以避免出現三連字）
            if (n >= 2 && res[n - 1] == c && res[n - 2] == c){
                continue;
            }
            res.push_back(c);
        }
        return res;
    }
};